### ISSUE TITLE
Keys.REPRESENTED_PLAYER not working with items

### ISSUE THREAD

**Issue by pie-flavor** [SpongePowered/Sponge#1406]
`Keys.REPRESENTED_PLAYER` on a skull item does not work.

Tested via the following code, executed on SpongeVanilla 1.12-7.0.0-BETA-288:
```java
Player player = Sponge.getServer().getPlayer("pie_flavor").get();
ItemStack stack = ItemStack.builder()
        .itemType(ItemTypes.SKULL)
        .keyValue(Keys.SKULL_TYPE, SkullTypes.PLAYER)
        .keyValue(Keys.REPRESENTED_PLAYER, player.getProfile())
        .build();
player.getInventory().offer(stack);
```
The item shows up as a bare Steve head, no extra data. Examination of the internals returned this (and nothing else):
```
{id:"minecraft:skull",Count:1b,Damage:3s}
```

**Comment by Zidane** [comment]
@Aaron1011 Can you fix this up since you did skulls?

**Comment by parlough** [comment]
@Zidane Was this not properly solved with #1409?

**Comment by Zidane** [comment]
Oh, completely spaced on this.
